File: app/domain/model/fizz_buzz_value.go
package model

// FizzBuzzValue 値オブジェクト（Value Object パターン）
type FizzBuzzValue struct {
	number int
	value  string
}

// NewFizzBuzzValue FizzBuzzValueのコンストラクタ
func NewFizzBuzzValue(number int, value string) *FizzBuzzValue {
	return &FizzBuzzValue{number: number, value: value}
}

// Number ゲッターメソッド
func (fv *FizzBuzzValue) Number() int {
	return fv.number
}

// Value ゲッターメソッド
func (fv *FizzBuzzValue) Value() string {
	return fv.value
}

// ToString 文字列表現を返す
func (fv *FizzBuzzValue) ToString() string {
	return fv.value
}

// Equal 等価性を比較
func (fv *FizzBuzzValue) Equal(other *FizzBuzzValue) bool {
	return fv.number == other.number && fv.value == other.value
}

File: app/src/Domain/Model/FizzBuzzValue.php
<?php

namespace App\Domain\Model;

use App\Assertions;

class FizzBuzzValue
{
    use Assertions;
    
    private int $number;
    private string $value;

    public function __construct(int $number, string $value)
    {
        $this->assert(fn() => $number >= 0, '数値は0以上である必要があります');
        $this->number = $number;
        $this->value = $value;
    }

    public function getNumber(): int
    {
        return $this->number;
    }

    public function getValue(): string
    {
        return $this->value;
    }

    public function __toString(): string
    {
        return $this->number . ':' . $this->value;
    }

    public function equals(FizzBuzzValue $other): bool
    {
        return $this->number === $other->number && $this->value === $other->value;
    }
}
